string[] str1
string[] str2
string[] str3
string[] str4
string[] str5